jina/peapods/zmq.py

```diff
@@ -316,15 +316,18 @@ def close(self):
             for s in self.opened_socks:
                 s.flush()
             super().close()
-            self.io_loop.stop()
-            # Replace handle events function, to skip
-            # None event after sockets are closed.
-            if hasattr(self.in_sock, '_handle_events'):
-                self.in_sock._handle_events = lambda *args, **kwargs: None
-            if hasattr(self.out_sock, '_handle_events'):
-                self.out_sock._handle_events = lambda *args, **kwargs: None
-            if hasattr(self.ctrl_sock, '_handle_events'):
-                self.ctrl_sock._handle_events = lambda *args, **kwargs: None
+            try:
+                self.io_loop.stop()
+                # Replace handle events function, to skip
+                # None event after sockets are closed.
+                if hasattr(self.in_sock, '_handle_events'):
+                    self.in_sock._handle_events = lambda *args, **kwargs: None
+                if hasattr(self.out_sock, '_handle_events'):
+                    self.out_sock._handle_events = lambda *args, **kwargs: None
+                if hasattr(self.ctrl_sock, '_handle_events'):
+                    self.ctrl_sock._handle_events = lambda *args, **kwargs: None
+            except AttributeError as e:
+                self.logger.info(f'failed to stop. {e}')
```

Comment:
what could possibly happen here?

Reply:
When the address conflict happens, these attributes will not be set and this leads to an exception  in closing the process and messed up the error messages.

> what could possibly happen here?

When the address conflict happens, these attributes will not be set and this leads to an exception  in closing the process and messed up the error messages.

Reply:
then it should be error not info